src/ServerDaemon.chpl

```diff
@@ -724,6 +736,30 @@ module ServerDaemon {
                                                                             elapsed));
             this.shutdown(); 
         }
+
+        /* While waiting for a message from client, do checkpointing if desired. */
+        proc startIdleServer() throws {
+          // Do nothing if the checkpointing module was not included.
+          if commandMap.contains("auto_checkpoint") {
```

Comment:
I have preference towards putting this and `needMemBasedCkpt` in a helper and calling that helper in the very top level `while` loop. Because it looks like we may end up calling a lot of functions that are no-ops. Function call is OK, but I think putting checks that may result in no-op in the top-level makes the code easier to read and maintain IMO.